python/rmm/tests/test_rmm.py

```diff
@@ -58,35 +59,20 @@ def test_rmm_modes(dtype, nelem, alloc, managed, pool):
     assert rmm.is_initialized()
 
     array_tester(dtype, nelem, alloc)
-
-
-def test_uninitialized():
-    rmm._finalize()
-    assert not rmm.is_initialized()
-    rmm.reinitialize()  # so further tests will pass
```

Comment:
If a user wants to explicitly free a memory pool, how do they do it now? We should have a test if possible.

Reply:
By calling `reinitialize()` -- we test this in `test_rmm_modes` above.